### PR TITLE
Remove feature flag checks from DLM serialisation

### PR DESCRIPTION
**Pull request by andreidan**
For DLM we didn't want to leave any trace on the wire as it is an experimental feature however, a rolling upgrade from a cluster without DLM to a cluster with DLM (e.g. `8.8.0` to `8.9.0-SNAPSHOT`) was not possible at the moment.

This removes the feature flag checks from the serialisation code for DLM and always read/write an optional writeable instead, **starting with `8.9.0`** and onwards

The serialisation cases we want to cover are mixed clusters of `8.8.x` and `8.9.0-SNAPSHOT`. We have two options:
- 8.8 has DLM disabled and 8.9 has it enabled (due to being a SNAPSHOT build)
- <s>both versions have DLM enabled</s> This is not the case anymore as we've disabled DLM serialisation in 8.8.1 and onwards with https://github.com/elastic/elasticsearch/pull/96623

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/metadata/DataLifecycle.java [hunk 1]
```diff
@@ -131,21 +131,14 @@ public int hashCode() {
...
     public DataLifecycle(StreamInput in) throws IOException {
         if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_007)) {
             dataRetention = in.readOptionalWriteable(Retention::read);
         } else {
-            var value = in.readOptionalTimeValue();
-            if (value == null) {
-                dataRetention = null;
-            } else {
-                dataRetention = new Retention(value);
-            }
```
Comment (andreidan) [inline server/src/main/java/org/elasticsearch/cluster/metadata/DataLifecycle.java:148]: As `8.8.0` is disabled (except for test clusters) I'm tempted to not add any special handling here

**Review by dakrone — APPROVED** [review]
LGTM assuming CI is happy

**Review by mark-vieira — APPROVED** [review]
LGTM

**Comment by andreidan** [comment]
@elasticmachine update branch

**Comment by andreidan** [comment]
@elasticmachine update branch

**Comment by andreidan** [comment]
Restarting tests due to https://github.com/elastic/elasticsearch/issues/96618 

@elasticmachine run elasticsearch-ci/part-2

File: server/src/main/java/org/elasticsearch/cluster/metadata/DataLifecycle.java (hunk 1 above)
Comment (gmarouli) [reply server/src/main/java/org/elasticsearch/cluster/metadata/DataLifecycle.java:148]: Same here? Assuming my understanding is correct I think this makes sense. This is a feature flag, so we do not have to provide guarantees for a cluster moving from `8.8.0` to `8.9.0`, so our main concern is the mixed tests that use the `8.8.1`? Right?

**Review by gmarouli — COMMENTED** [review]
LGTM, thanks for working on this @andreidan , that was quite a puzzle 🚀

**Comment by gmarouli** [comment]
> @gmarouli `8.8.0` has DLM disabled as it's a stable release and #96391 was reverted. `8.8.1-SNAPSHOT` (and all subsequent 8.8.x release, where x >) has DLM disabled due to #96623

I see, since we do not add the DLM flag to these tests the only way to enable DLM is via the snapshot build. So, we are "safe". Clear.

**Pull request merged**